--- libdreamdvd/main.c.orig	2012-12-14 22:31:17.324074075 +0200
+++ libdreamdvd/main.c	2012-12-14 22:31:19.276083752 +0200
@@ -28,6 +28,10 @@
 #include "main.h"
 #include "mpegaudioenc.h"
 #include "a52dec.h"
+#if defined(__sh__)
+//activate LPCM support
+#define HARDWARE_SUPPORT_LPCM
+#endif
 
 #define Debug(level, str, ...) (DebugLevel > level ? printf("LIBDVD: %07.3f: " str, (float) ddvd_get_time() / 1000.0, ##__VA_ARGS__) : 0)
 
@@ -44,6 +48,9 @@
 
 	while (written < count) {
 		n = write(fd, &ptr[written], count - written);
+#if defined(__sh__)
+		return 0;
+#else
 		if (n < 0) {
 			if (errno != EINTR) {
 				perror("LIBDVD: write");
@@ -52,6 +59,7 @@
 		}
 		else
 			written += n;
+#endif
 	}
 
 	return written;
@@ -129,7 +137,11 @@
 	// defaults
 	ddvd_set_ac3thru(pconfig, 0);
 	ddvd_set_language(pconfig, "en");
+#if defined(__sh__)
+	ddvd_set_dvd_path(pconfig, "/dev/sr0");
+#else
 	ddvd_set_dvd_path(pconfig, "/dev/cdroms/cdrom0");
+#endif
 	ddvd_set_video(pconfig, DDVD_4_3, DDVD_LETTERBOX, DDVD_PAL);
 	ddvd_set_lfb(pconfig, NULL, 720, 576, 1, 720);
 	struct ddvd_resume resume_info;
@@ -611,8 +623,9 @@
 	enum ddvd_result res = DDVD_OK;
 	int msg;
 	// try to load liba52.so.0 for softdecoding
+#if !defined(__sh__)
 	int have_liba52 = ddvd_load_liba52();
-
+#endif
 	// decide which resize routine we should use
 	// on 4bpp mode we use bicubic resize for sd skins because we get much better results with subtitles and the speed is ok
 	// for hd skins we use nearest neighbor resize because upscaling to hd is too slow with bicubic resize
@@ -795,11 +808,12 @@
 	ddvd_mpa_init(48000, 192000);	//init MPA Encoder with 48kHz and 192k Bitrate
 
 	int ac3thru = 1;
+#if !defined(__sh__)
 	if (have_liba52) {
 		state = a52_init(0);	//init AC3 Decoder
 		ac3thru = playerconfig->ac3thru;
 	}
-
+#endif
 	char osdtext[512];
 	osdtext[0] = 0;
 
@@ -878,8 +892,13 @@
 		goto err_dvdnav_open;
 	}
 
+#if defined(__sh__)
+	/* set read ahead cache usage to yes for ufs910 */
+	if (dvdnav_set_readahead_flag(dvdnav, 1) != DVDNAV_STATUS_OK) {
+#else
 	/* set read ahead cache usage to no */
 	if (dvdnav_set_readahead_flag(dvdnav, 0) != DVDNAV_STATUS_OK) {
+#endif
 		Debug(1, "Error on dvdnav_set_readahead_flag: %s\n", dvdnav_err_to_string(dvdnav));
 		res = DDVD_FAIL_PREFS;
 		goto err_dvdnav;
@@ -1010,7 +1029,11 @@
 				ddvd_spu_play = ddvd_spu_ind; // skip remaining subtitles
 			}
 
+#if defined(__sh__)
+			result = dvdnav_get_next_cache_block(dvdnav, &buf, &event, &len);
+#else
 			result = dvdnav_get_next_block(dvdnav, buf, &event, &len);
+#endif
 			if (result == DVDNAV_STATUS_ERR) {
 				Debug(1, "Error getting next block: %s\n", dvdnav_err_to_string(dvdnav));
 				sprintf(osdtext, "Error: Getting next block: %s", dvdnav_err_to_string(dvdnav));
@@ -1262,10 +1285,22 @@
 					else if ((buf[14 + 3]) == 0xC0 + audio_id) {	// mpeg audio
 						if (audio_type != DDVD_MPEG) {
 							//Debug(1, "Switch to MPEG Audio\n");
+#if defined(__sh__)
+							//stop audio bevor change encoding
+							if (ioctl(ddvd_fdaudio, AUDIO_STOP) < 0)
+								perror("LIBDVD: AUDIO_STOP");
+							if (ioctl(ddvd_fdaudio, AUDIO_CLEAR_BUFFER) < 0)
+								perror("LIBDVD: AUDIO_CLEAR_BUFFER");
+#endif
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_AV_SYNC, 1) < 0)
 								perror("LIBDVD: AUDIO_SET_AV_SYNC");
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_BYPASS_MODE, 1) < 0)
 								perror("LIBDVD: AUDIO_SET_BYPASS_MODE");
+#if defined(__sh__)
+							//start audio after encoding set
+							if (ioctl(ddvd_fdaudio, AUDIO_PLAY) < 0)
+								perror("LIBDVD: AUDIO_PLAY");
+#endif
 							audio_type = DDVD_MPEG;
 						}
 
@@ -1292,10 +1327,22 @@
 
 						if (audio_type != DDVD_LPCM) {
 							//Debug(1, "Switch to LPCM Audio\n");
+#if defined(__sh__)
+							//stop audio bevor change encoding
+							if (ioctl(ddvd_fdaudio, AUDIO_STOP) < 0)
+								perror("LIBDVD: AUDIO_STOP");
+							if (ioctl(ddvd_fdaudio, AUDIO_CLEAR_BUFFER) < 0)
+								perror("LIBDVD: AUDIO_CLEAR_BUFFER");
+#endif
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_AV_SYNC, 1) < 0)
 								perror("LIBDVD: AUDIO_SET_AV_SYNC");
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_BYPASS_MODE, lpcm_mode) < 0)
 								perror("LIBDVD: AUDIO_SET_BYPASS_MODE");
+#if defined(__sh__)
+							//start audio after encoding set
+							if (ioctl(ddvd_fdaudio, AUDIO_PLAY) < 0)
+								perror("AUDIO_PLAY");
+#endif
 							audio_type = DDVD_LPCM;
 							ddvd_lpcm_count = 0;
 						}
@@ -1360,6 +1407,13 @@
 					else if ((buf[14 + 3]) == 0xBD && (buf[14 + buf[14 + 8] + 9]) == 0x88 + audio_id) {	// dts audio
 						if (audio_type != DDVD_DTS) {
 							//Debug(1, "Switch to DTS Audio (thru)\n");
+#if defined(__sh__)
+							//stop audio bevor change encoding
+							if (ioctl(ddvd_fdaudio, AUDIO_STOP) < 0)
+								perror("LIBDVD: AUDIO_STOP");
+							if (ioctl(ddvd_fdaudio, AUDIO_CLEAR_BUFFER) < 0)
+								perror("LIBDVD: AUDIO_CLEAR_BUFFER");
+#endif
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_AV_SYNC, 1) < 0)
 								perror("LIBDVD: AUDIO_SET_AV_SYNC");
 #ifdef CONVERT_TO_DVB_COMPLIANT_DTS
@@ -1368,6 +1422,11 @@
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_BYPASS_MODE, 5) < 0)	// DTS VOB
 #endif
 								perror("LIBDVD: AUDIO_SET_BYPASS_MODE");
+#if defined(__sh__)
+							//start audio after encoding set
+							if (ioctl(ddvd_fdaudio, AUDIO_PLAY) < 0)
+								perror("AUDIO_PLAY");
+#endif
 							audio_type = DDVD_DTS;
 						}
 
@@ -1397,7 +1456,11 @@
 						if (audio_type != DDVD_AC3) {
 							//Debug(1, "Switch to AC3 Audio\n");
 							int bypassmode;
+#if defined(__sh__)
+							if (ac3thru)
+#else
 							if (ac3thru || !have_liba52) // !have_liba52 and !ac3thru should never happen, but who knows ;)
+#endif
 #ifdef CONVERT_TO_DVB_COMPLIANT_AC3
 								bypassmode = 0;
 #else
@@ -1405,10 +1468,22 @@
 #endif
 							else
 								bypassmode = 1;
+#if defined(__sh__)
+							//stop audio bevor change encoding
+							if (ioctl(ddvd_fdaudio, AUDIO_STOP) < 0)
+								perror("LIBDVD: AUDIO_STOP");
+							if (ioctl(ddvd_fdaudio, AUDIO_CLEAR_BUFFER) < 0)
+								perror("LIBDVD: AUDIO_CLEAR_BUFFER");
+#endif
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_AV_SYNC, 1) < 0)
 								perror("LIBDVD: AUDIO_SET_AV_SYNC");
 							if (ioctl(ddvd_fdaudio, AUDIO_SET_BYPASS_MODE, bypassmode) < 0)
 									perror("LIBDVD: AUDIO_SET_BYPASS_MODE");
+#if defined(__sh__)
+							//start audio after encoding set
+							if (ioctl(ddvd_fdaudio, AUDIO_PLAY) < 0)
+								perror("AUDIO_PLAY");
+#endif
 							audio_type = DDVD_AC3;
 						}
 
@@ -1421,8 +1496,14 @@
 							apts |= (buf[14 + 13] >> 1);
 							//Debug(1, "APTS=%X\n",(int)apts);
 						}
-
+#if defined(__sh__)
+						if (ac3thru) {
+#else
 						if (ac3thru || !have_liba52) {	// !have_liba52 and !ac3thru should never happen, but who knows ;)
+#endif
+#if defined(__sh__)
+						safe_write(ddvd_ac3_fd, buf + 14, buf[19] + (buf[18] << 8) + 6);
+#else
 #ifdef CONVERT_TO_DVB_COMPLIANT_AC3
 							unsigned short pes_len = (buf[14 + 4] << 8 | buf[14 + 5]);
 							pes_len -= 4;	// strip first 4 bytes of pes payload
@@ -1434,6 +1515,7 @@
 #else
 							safe_write(ddvd_ac3_fd, buf + 14, buf[19] + (buf[18] << 8) + 6);
 #endif
+#endif
 							//fwrite(buf + buf[22] + 27, 1, ((buf[18] << 8) | buf[19]) - buf[22] - 7, fac3); //debugwrite
 						}
 						else {
@@ -2695,12 +2777,12 @@
 err_open_fdvideo:
 	close(ddvd_output_fd);
 err_open_output_fd:
-
+#if !defined(__sh__)
 	if (have_liba52) {
 		a52_free(state);
 		ddvd_close_liba52();
 	}
-
+#endif
 	//Clear Screen
 	blit_area.x_start = blit_area.y_start = 0;
 	blit_area.x_end = ddvd_screeninfo_xres - 1;
@@ -3043,6 +3125,14 @@
 }
 
 // blit to argb in 32bit mode
+#if defined(__sh__)
+#define alpha_composite(composite, fg, alpha, bg) {  \
+    unsigned short temp = ((unsigned short)(fg)*(unsigned short)(alpha) +   \
+    (unsigned short)(bg)*(unsigned short)(255 - (unsigned short)(alpha)) +  \
+    (unsigned short)128);  \
+    (composite) = (unsigned char)((temp + (temp >> 8)) >> 8);  \
+}
+#endif
 static void ddvd_blit_to_argb(void *_dst, const void *_src, int pix)
 {
 	unsigned long *dst = _dst;
@@ -3050,14 +3140,27 @@
 	while (pix--) {
 		int p = (*src++);
 		int a, r, g, b;
+#if defined(__sh__)
+		int r1, g1, b1;
+#endif
 		if (p == 0) {
 			r = g = b = a = 0;	//clear screen (transparency)
 		}
 		else {
+#if defined(__sh__)
+			a = 0xFF - (ddvd_tr[p] >> 8);
+			r1 = ddvd_rd[p] >> 8;
+			g1 = ddvd_gn[p] >> 8;
+			b1 = ddvd_bl[p] >> 8;
+			alpha_composite(r, r1, a, 0);
+			alpha_composite(g, g1, a, 0);
+			alpha_composite(b, b1, a, 0);
+#else
 			a = 0xFF - (ddvd_tr[p] >> 8);
 			r = ddvd_rd[p] >> 8;
 			g = ddvd_gn[p] >> 8;
 			b = ddvd_bl[p] >> 8;
+#endif
 		}
 		*dst++ = (a << 24) | (r << 16) | (g << 8) | (b << 0);
 	}
