--- stm-target-gcc.spec.orig	2012-10-27 22:04:27.208194064 +0200
+++ stm-target-gcc.spec	2012-10-27 22:04:27.172194140 +0200
@@ -15,6 +15,7 @@
 Source0: ftp://gcc.gnu.org/pub/gcc/releases/gcc-%{version}/gcc-%{version}.tar.bz2
 Source1: gcc-protoize.1
 
+Patch0: stm-target-gcc.%{version}-%{release}.diff
 Patch1: gcc-4.2.4-multilibpath.patch
 Patch3: gcc-4.5.0-autoconf.patch
 
@@ -168,6 +169,7 @@
 
 %prep
 %setup -n gcc-%{version} -q
+%patch0 -p1
 
 # Common patches
 %if "%{_stm_target_cpu}" != "arm"
@@ -251,8 +253,6 @@
 export CFLAGS="-Os"
 export CXXFLAGS="-Os"
 
-export CC_FOR_BUILD="gcc -m32"
-export CXX_FOR_BUILD="g++ -m32"
 
 # When cross compiling this fails because it needs to run code
 export ac_cv_header_stdc=yes
@@ -388,6 +387,8 @@
 rm -r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_man_dir}/man7
 rm -r %{buildroot}%{_gccdir}/install-tools
 rm -r %{buildroot}%{_stm_cross_target_dir}%{_stm_target_libexec_dir}/gcc/%{_stm_target_config}/%{version}/install-tools
+rm -rf %{buildroot}%{_gccdir}/include/ssl
+rm -rf %{buildroot}%{_gccdir}/include/xulrunner-1.9.2.17/nss
 
 %unfixincludes %{buildroot}%{_gccdir}
 
