--- stm-cross-binutils.spec.orig	2013-02-26 03:05:03.000000000 +0100
+++ stm-cross-binutils.spec	2013-03-20 21:37:54.363059014 +0100
@@ -39,9 +39,6 @@
 BuildRequires: texinfo
 %endif
 BuildRequires: %{_stm_pkg_prefix}-host-autoconf
-BuildRequires: %{_stm_pkg_prefix}-host-bison
-BuildRequires: %{_stm_pkg_prefix}-host-flex
-BuildRequires: %{_stm_pkg_prefix}-host-gettext
 BuildRequires: %{_stm_pkg_prefix}-host-rpmconfig
 
 Group: Development/Languages
@@ -117,6 +114,7 @@
        --enable-nls \
        --disable-multilib \
        --with-system-zlib \
+       --disable-werror \
 %if "%{_stm_target_fpu}" == "yes"
 %if %target_arch armv7 armv7_uclibc
         --with-float=hard \
